### ISSUE TITLE
Virtual Piano freezes sporadically

### ISSUE THREAD

**Issue by fsgregs** [Jojo-Schmitz/MuseScore#681]
JoJo:  Hope this reaches you.  In using MS 3.70, I always launch the virtual piano menu with the "p" key.  It appears fine and works fine ... until it doesn't.  Somewhere in the score I am working on, the piano keys will freeze in the menu.  My cursor can be placed anywhere ... the song can continue playing, but the piano keys refuse to change to follow the cursor.

Here is an example:  Note that the piano keys are all light blue lit up in the key of A7, but the actual cursor is in the "Gm6" measure.  The keys in the virtual piano should have changed with them.

As to reproduction, that is a mystery.  Sometimes the bug appears ... sometimes it doesn't. Load up a song, turn on the virtual keyboard, click around the score in random measures, and see if the bug manifests.  Note that I am using key chords in the song (Ctrl K).  Could that be part of the problem?

  
![image](https://github.com/user-attachments/assets/318866cb-87c5-4413-80cf-784d4ef53931)

**Comment by Jojo-Schmitz** [comment]
And this not the case with 3.6.2?

**Comment by Anonym271** [comment]
Yes, I can confirm this. I also have this problem with MuseScore 3.7, it usually occurs within 10 or 15 minutes after start in my case. The Piano Keyboard suddenly freezes. Hiding and showing it again causes it to update the pressed notes to the current state but it will still not update when something changes as soon as the bug occurs, restart seems the only fix.

I could not observe this problem within about 1 hour of testing in MS 3.6.2 and I am also pretty sure that I never had this problem before updating to MS 3.7.

Using Windows 11.

**Comment by lyrra** [comment]
I've bisected this to commit ad179187

**Comment by Jojo-Schmitz** [comment]
Excellent job, thanks!
@Anonym271 is your commit https://github.com/Anonym271/MuseScore/commit/77c62180e2d9fcc10965628c9c0167e7751a521f supposed to fix this?

**Comment by Anonym271** [comment]
Yes, it seems to be working in my build now after these changes.

**Comment by Jojo-Schmitz** [comment]
OK, I'll turn it into a PR ;-)

**Comment by Jojo-Schmitz** [comment]
@lyrra, @fsgregs can you test the artifacts from #921?

**Comment by lyrra** [comment]
> [@lyrra](https://github.com/lyrra), [@fsgregs](https://github.com/fsgregs) can you test the artifacts from [#921](https://github.com/Jojo-Schmitz/MuseScore/pull/921)?

Confirming that the patch works. (without the patch, the piano-keyboard freezes after around 10-20 min).
Great work!
